modules/payment/src/Entity/PaymentGateway.php

```diff
@@ -159,4 +161,67 @@ protected function getPluginCollection() {
     return $this->pluginCollection;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function postSave(EntityStorageInterface $storage, $update = TRUE) {
+    parent::postSave($storage, $update);
+
+    $bundle_of = $this->getEntityType()->getBundleOf();
+    if (!$update) {
+      \Drupal::getContainer()->get('entity_bundle.listener')->onBundleCreate($this->id(), $bundle_of);
+    }
+    else {
+      $entity_field_manager = \Drupal::getContainer()->get('entity_field.manager');
```

Comment:
Can't we do this static and coupled ::getContainer() using a decoupled dependency injection? Or am I crazy?